backend/fms_core/viewsets/library.py

```diff
@@ -0,0 +1,117 @@
+        {
+            "name": "Add Experiments",
+            "description": "Upload the provided template with experiment run information.",
+            "template": [EXPERIMENT_MGI_TEMPLATE["identity"]],
+            "importer": ExperimentRunImporter,
+        },
+    ]
+
+    template_prefill_list = [
+        {"template": EXPERIMENT_MGI_TEMPLATE},
+    ]
+
+    def get_queryset(self):
+        container_barcode = self.request.query_params.get('container__barcode__recursive')
+        container_name = self.request.query_params.get('container__name__recursive')
+        recursive = container_barcode or container_name
+
+        if recursive:
+            containers = Container.objects.all()
+            if container_barcode:
+                containers = containers.filter(barcode=container_barcode)
+            if container_name:
+                containers = containers.filter(name=container_name)
+
+            container_ids = tuple(containers.values_list('id', flat=True))
+
+            if not container_ids:
+                container_ids = tuple([None])
+
+            parent_containers = Container.objects.raw('''WITH RECURSIVE parent(id, location_id) AS (
+                                                                   SELECT id, location_id
+                                                                   FROM fms_core_container
+                                                                   WHERE id IN %s
+                                                                   UNION ALL
+                                                                   SELECT child.id, child.location_id
+                                                                   FROM fms_core_container AS child, parent
+                                                                   WHERE child.location_id = parent.id
+                                                               )
+                                                               SELECT * FROM parent''', params=[container_ids])
+
+            return self.queryset.filter(container__in=parent_containers)
+
+        return self.queryset
+
+    @action(detail=False, methods=["get"])
+    def list_export(self, _request):
+        serializer = LibraryExportSerializer(self.filter_queryset(self.get_queryset()), many=True)
+        return Response(serializer.data)
+
+    def get_renderer_context(self):
+        context = super().get_renderer_context()
+        if self.action == 'list_export':
+            fields = LibraryExportSerializer.Meta.fields
+            context['header'] = fields
+            context['labels'] = {i: i.replace('_', ' ').capitalize() for i in fields}
+        return context
+
+    @action(detail=False, methods=["get"])
+    def summary(self, _request):
+        """
+        Returns summary statistics about the current set of libraries in the
+        database.
+        """
+        return Response({"total_count": self.get_queryset().all().count()})
```

Comment:
I would add a breakdown by library_type (formatted like the process_measurement "protocol_count" summary).